### PR TITLE
Fix bug with SessionId usage for checkout

### LINKED ISSUE #92: Using SessionId for checkout errors due
**Describe the bug**

My checkout use-case is calling Stripe with a `sessionId`. When calling the `redirectToCheckout` method with a `sessionId` no other values can be present in the request; if using `sessionId` to redirect then only `sessionId` can be in the object. 

**To Reproduce**
1. Set up a small example where you try to use a sessionId to checkout.
2. Click on your button, it should fail, and then look at the console. You should see this:
> IntegrationError: stripe.redirectToCheckout: Do not provide other parameters when providing sessionId. Specify all parameters on your server when creating the CheckoutSession.

**Expected behavior**

I would expect that if the method of redirecting to checkout is `sessionId` that the object provided to the `redirectToCheckout` method would only include the `sessionId`. 

Am I missing something here? I'm still learning Stripe, Vue, and javascript so there's good chance I'm misundrstanding how this should work. Here's a very small example that replicates my issue. Have others been using this project to do checkouts with sessionIds? Thanks!

```vue
<template>
    <stripe-checkout
            ref="checkoutRef"
            :pk="publishableKey"
            :sessionId="sessionId"

    >
        <template slot="checkout-button">
            <button @click="checkout">Shutup and take my money!</button>
        </template>
    </stripe-checkout>
</template>

<script>
    import { StripeCheckout } from 'vue-stripe-checkout';
    export default {
        components: {
            StripeCheckout
        },
        data: () => ({
            loading: false,
            publishableKey: "pk_test_rbM3yLhvn31PWOixMsLrYUFT0065xhosZl",//process.env.PUBLISHABLE_KEY,
            sessionId: 'cs_test_MmNYSvafLSYZRrhthcGxOQUvNduAhx9MPFMvbhUUTL9iruH6sVhN70kD',
        }),
        methods: {
            checkout () {
                this.$refs.checkoutRef.redirectToCheckout();
            }
        }
    }
</script>
```

### PR DESCRIPTION
**Pull request by tbrown1979**
Seems to fix #92 

This seemed like a relatively quick fix to the problem I was having. Seems to work for me just fine. Let me know if there are any changes you'd like to see.

### CODE REVIEW

File: src/Checkout.vue [hunk 1]
```diff
@@ -60,23 +60,29 @@ export default {
...
-          });
+          if (this.sessionId == null) {
+            stripe.redirectToCheckout({
+              billingAddressCollection: this.billingAddressCollection,
+              cancelUrl: this.cancelUrl,
+              clientReferenceId: this.clientReferenceId,
+              customerEmail: this.customerEmail,
+              items: this.items,
+              locale: this.locale,
+              sessionId: this.sessionId,
```
Comment (jofftiquez) [inline src/Checkout.vue]: Is this still necessary if the `sessionId` prop is null?

Comment (tbrown1979) [reply src/Checkout.vue]: Nope, good catch 😄

**Comment by tbrown1979** [comment]
@jofftiquez Yep, that's exactly right! Creating the Checkout on the backend is a necessity for my use-case, so I have to make use of this `sessionId` field.

File: src/Checkout.vue (hunk 1 above)
Comment (jofftiquez) [reply src/Checkout.vue]: @tbrown1979 kindly remove the `sessionId` key and update na PR. Thanks :)

**Comment by jofftiquez** [comment]
@tbrown1979 Right. I see. Just update the PR and I'll see what else I can update so we can merge it later. Thanks.

**Comment by jofftiquez** [comment]
Oh you already updated the PR. Alright then. Let me just check it. Will merge soon.

**Comment by platform-kit** [comment]
@jofftiquez / @tbrown1979 , how do you use the session ID? Would love to see an example added to the readme.

**Comment by jofftiquez** [comment]
Hello @platform-kit would you mind creating a separate issue for this? I would love to add an example sometime in the future. 

To answer you question, here's a pseudo code on how you use a session in Stripe.

1. In your backend create a session with the products, or skus you want to purchase or subscribe to.
2. Stripe will create a session for you, grab the sessionId of that session.
3. Use the sessionId from the backend to create a checkout in frontend.

**Comment by platform-kit** [comment]
@jofftiquez Thanks for the quick response. I've already my backend returning a session ID, just wondering how to pass it into your component. I'll poke around... if I can't come up with a solution I'll create a new issue.

**Comment by jofftiquez** [comment]
You're right. There's nothing about sessionId are mentioned in the docs. 

The checkout component actually has a `sessionId` property, [check here](https://github.com/jofftiquez/vue-stripe-checkout/blob/e968fda20c25ca3762b5c8e223727d6df2e3004d/src/Checkout.vue#L51). You can pass the session id here like so:

```vue
<stripe-checkout :pk="pk" :sessionId="yourSessionId"  />
```

The only required props are `pk` and `sessionId` if you are doing checkout using sessions.

**Comment by platform-kit** [comment]
That was the missing link! Thanks for your help.

**Pull request merged**